internal/mdxc/application_identity.go

```diff
@@ -196,11 +196,14 @@ type applicationIdentityFunctionGCP func(context.Context, *gcp.ApplicationIdenti
 
 func convertApplicationIdentityConfigTerraformToGCP(d *ApplicationIdentityData, a *gcp.ApplicationIdentityConfig, c *gcp.GCPConfig) {
 	a.ID = d.Id.Value
+	a.ServiceAccountEmail = d.Id.Value
```

Comment:
Since we're using the email as the id, we may be able to remove this field from the appIdentityConfig